dimod/include/dimod/quadratic_model.h

```diff
@@ -474,6 +617,81 @@ class QuadraticModelBase {
     std::vector<Neighborhood<bias_type, index_type>> adj_;
 
     bias_type offset_;
+
+    friend class ConstQuadraticIterator<Bias, Index>;
+};
+
+template <class Bias, class Index>
+class ConstQuadraticIterator {
```

Comment:
I am not the user of this class but I do not know how much it helps, having to remember the internal working of this class seems like a hard thing for me, I would just iterate over the neighborhood using the other tools above.